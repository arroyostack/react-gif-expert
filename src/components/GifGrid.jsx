import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs"
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
    const { images, isLoading } = useFetchGifs(category)

    return (
        <>
            <p>{category}</p>

            <h1>{isLoading && 'Loading...'}</h1>
            <div className="card-grid">
                {images.map((image) => {
                    return <GifItem
                        key={image.id}
                        {...image}
                    />
                })}
            </div>

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}