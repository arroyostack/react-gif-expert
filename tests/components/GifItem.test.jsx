import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Unit testing for <GifItem/>', () => {
    const title = 'Superman';
    const url = 'https://one-punch.com/superman'

    // TEST 1
    test('should match the snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    })
    //  TEST 4
    test('should show the image with adecuated url and alt', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug()
        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( alt );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    })
    // TEST 3
    test('should show the title on the component', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy()
    })
})