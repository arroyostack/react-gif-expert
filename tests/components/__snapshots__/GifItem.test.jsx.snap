// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing for <GifItem/> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Superman"
      src="https://one-punch.com/superman"
    />
    <h6>
      Superman
    </h6>
  </div>
</div>
`;
